storm-core/src/jvm/backtype/storm/scheduler/resource/strategies/priority/ISchedulingPriorityStrategy.java

```diff
@@ -16,22 +16,26 @@
  * limitations under the License.
  */
 
-package backtype.storm.scheduler.resource.strategies;
-
-import java.util.Collection;
-import java.util.Map;
+package backtype.storm.scheduler.resource.strategies.priority;
 
+import backtype.storm.scheduler.Cluster;
 import backtype.storm.scheduler.Topologies;
-import backtype.storm.scheduler.ExecutorDetails;
 import backtype.storm.scheduler.TopologyDetails;
-import backtype.storm.scheduler.WorkerSlot;
-import backtype.storm.scheduler.resource.RAS_Node;
+import backtype.storm.scheduler.resource.RAS_Nodes;
+import backtype.storm.scheduler.resource.User;
 
-/**
- * An interface to for implementing different scheduling strategies for the resource aware scheduling
- * In the future stategies will be pluggable
- */
-public interface IStrategy {
+import java.util.Map;
+
+public interface ISchedulingPriorityStrategy {
+
+    /**
+     * initializes
+     */
+    public void prepare(Topologies topologies, Cluster cluster, Map<String, User> userMap, RAS_Nodes nodes);
```

Comment:
Ya but just like my comment in for the IEviction interface.  The DefaultSchedulingPriorityStrategy might not use those variables but someone might implement a strategy in the future that might.  And we don't want to have to come back and add those parameters in the future which would break backwards compatibility